src/main/java/tasks/Task.java

```diff
@@ -0,0 +1,125 @@
+package tasks;
+
+import utilities.ResponseHandler;
+
+public class Task {
+    private String taskName;
+    private boolean taskDone = false;
+    private String typeOfTask;
+
+    private String[] dateFromAndTo = new String[] {"NA", "NA"};
+    private String[] timeFromAndTo = new String[] {"NA", "NA"};
+
+    public Task(String taskName) {
+        this.taskName = taskName;
+    }
+
+    public Task(String taskName, String typeOfTask) {
+        this.taskName = taskName;
+        this.typeOfTask = typeOfTask;
+    }
+
+    /**
+     * Gets the name of the task.
+     *
+     * @return The name of the task.
+     */
+    public String getTaskName() {
+        return this.taskName;
+    }
+
+    /**
+     * Gets the type of the task.
+     *
+     * @return The type of the task.
+     */
+    public String getTaskType() {
+        return this.typeOfTask;
+    }
+
+    /**
+     * Checks if the task is marked as done.
+     *
+     * @return {@code true} if the task is marked as done, {@code false} otherwise.
+     */
+    public boolean isTaskDone() {
+        return this.taskDone;
+    }
+
+    /**
+     * Gets the status icon representing whether the task is done or not.
+     *
+     * @return The status icon ("X" if the task is done, " " (space) otherwise).
+     */
+    public String getStatusIcon() {
+        return (this.taskDone ? "X" : " ");
```

Comment:
I like how you used a ternary operator instead of an if else block to make your code seem neater.